src/main/java/teammates/ui/controller/InstructorFeedbackAddAction.java

```diff
@@ -146,33 +149,54 @@ private void createTemplateFeedbackQuestions(String courseId, String feedbackSes
         return new ArrayList<FeedbackQuestionAttributes>();
     }
 
-    private FeedbackSessionAttributes extractFeedbackSessionData() {
+    public FeedbackSessionAttributes extractFeedbackSessionData() {
         //TODO assert parameters are not null then update test
         //TODO make this method stateless
-        
+      
         FeedbackSessionAttributes newSession = new FeedbackSessionAttributes();
         newSession.setCourseId(getRequestParamValue(Const.ParamsNames.COURSE_ID));
         newSession.setFeedbackSessionName(Sanitizer.sanitizeTitle(
                 getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME)));
         
-        newSession.setCreatedTime(new Date());
+        // getting the class name from which it's being called 
+        String className = new Exception().getStackTrace()[1].getClassName();
+        
+        if (className == "InstructorFeedbackEditSaveAction") {
+            newSession.setCreatorEmail(getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_CREATOR));
+        } else {
+            newSession.setCreatedTime(new Date());            
+        }
+        
         newSession.setStartTime(TimeHelper.combineDateTime(
                 getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTDATE),
                 getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTTIME)));
```

Comment:
Using hardcoded class names also couples the implementation of this class to its callers. For example, if the class's name changes, the method doesn't do the right thing.